matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/OlmMachine.kt

```diff
@@ -306,13 +308,12 @@ internal class OlmMachine(
             }
 
             val devices =
-                    DeviceLists(
-                            deviceChanges?.changed ?: listOf(), deviceChanges?.left ?: listOf())
+                    DeviceLists(deviceChanges?.changed.orEmpty(), deviceChanges?.left.orEmpty())
             val adapter =
                     MoshiProvider.providesMoshi().adapter(ToDeviceSyncResponse::class.java)
-            val events = adapter.toJson(toDevice ?: ToDeviceSyncResponse())!!
+            val events = toDevice?.let { adapter.toJson(it) } ?: "[]"
```

Comment:
Are we sure we want to do this? This will, if serialization fails, create an empty list of to-device events.

Those events will potentially have room keys in them, we might want to know that this happened instead of silently failing, no?

Reply:
yeah will let it. Not sure it will ever happen anyhow

Reply:
It should never happen, but we still might want to either assert that it won't happen or at least log so if it does.

Reply:
done